pkg/server/event_handler.go

```diff
@@ -234,33 +240,57 @@ func (h *EventHandler[T, P]) HTTPHandler() http.Handler {
 // [GCS notification]: https://cloud.google.com/storage/docs/pubsub-notifications#format
 func (h *EventHandler[T, P]) Handle(ctx context.Context, m pubsub.Message) error {
 	logger := logging.FromContext(ctx)
+
+	eventBytes, processErr := h.generatePmapEventBytes(ctx, m)
+
+	attr := map[string]string{}
+
+	if processErr != nil {
+		if errors.Is(processErr, pmaperrors.ErrPubsubRetryable) {
+			return processErr
+		}
+		attr[PubSubAttrKeyNonRetryableErr] = processErr.Error()
+		logger.Errorw(processErr.Error(), "bucketId", m.Attributes["bucketId"], "objectId", m.Attributes["objectId"])
+		if err := h.failureMessenger.Send(ctx, eventBytes, attr); err != nil {
+			return fmt.Errorf("failed to send failure event downstream: %w", err)
+		}
+		return nil
+	}
+	if err := h.successMessenger.Send(ctx, eventBytes, attr); err != nil {
+		return fmt.Errorf("failed to send succuss event downstream: %w", err)
+	}
+	return nil
+}
+
+func (h *EventHandler[T, P]) generatePmapEventBytes(ctx context.Context, m pubsub.Message) ([]byte, error) {
+	// TODO(#20): The currently logic is to treat all errors that are not caused by processor as retryable.
+	// and all errors that are caused by processor as non-retryable.
+	// Therefore, we intend to keep this logic here, and do a more detailed differentiate later.
```

Comment:
```suggestion
	// TODO(#20): The previous logic is to treat all errors that are not caused by processor as retryable.
	// and all errors that are caused by processor as non-retryable.
	// We will keep this behavior and follow up to properly differentiate errors.
```

Reply:
Although... what will happen if we **not** make these retryable errors?

Reply:
I am making changes to make them non-retryable now. Are doing test on it.

Reply:
Just made all non-retryable. And also joined errors you mention in [here](https://github.com/abcxyz/pmap/pull/114#discussion_r1228830223) and [here](https://github.com/abcxyz/pmap/pull/114#discussion_r1228830254) to processErr. I can change it if we decide we only want user facing errors.